src/cli/cli_tcp.cpp

```diff
@@ -74,13 +102,80 @@ template <> otError TcpExample::Process<Cmd("init")>(Arg aArgs[])
     }
     else
     {
+#if OPENTHREAD_CONFIG_TLS_ENABLE
+        if (aArgs[0] == "tls")
+        {
+            mUseCircularSendBuffer = true;
+            mUseTLS                = true;
+
+            // mbedtls_debug_set_threshold(0);
+
+            mbedtls_ctr_drbg_init(&mCtrDrbg);
+            mbedtls_x509_crt_init(&mSrvCert);
+            mbedtls_pk_init(&mPKey);
+
+            mbedtls_ssl_init(&mSSLContext);
+            mbedtls_ssl_config_init(&mSSLConfig);
+            mbedtls_ssl_conf_rng(&mSSLConfig, mbedtls_ctr_drbg_random, &mCtrDrbg);
+            // mbedtls_ssl_conf_dbg(&mSSLConfig, MbedTlsDebugOutput, this);
+            mbedtls_ssl_conf_authmode(&mSSLConfig, MBEDTLS_SSL_VERIFY_NONE);
+            mbedtls_entropy_init(&mEntropy);
+
+#if (MBEDTLS_VERSION_NUMBER >= 0x03020000)
+            mbedtls_ssl_conf_min_tls_version(&mSSLConfig, MBEDTLS_SSL_VERSION_TLS1_2);
+            mbedtls_ssl_conf_max_tls_version(&mSSLConfig, MBEDTLS_SSL_VERSION_TLS1_2);
+#else
+            mbedtls_ssl_conf_min_version(&mSSLConfig, MBEDTLS_SSL_MAJOR_VERSION_3, MBEDTLS_SSL_MINOR_VERSION_3);
+            mbedtls_ssl_conf_max_version(&mSSLConfig, MBEDTLS_SSL_MAJOR_VERSION_3, MBEDTLS_SSL_MINOR_VERSION_3);
+#endif
+
+#if (MBEDTLS_VERSION_NUMBER >= 0x03000000)
+#include "crypto/mbedtls.hpp"
+            int rv = mbedtls_pk_parse_key(&mPKey, reinterpret_cast<const unsigned char *>(sSrvKey), sSrvKeyLength,
+                                          nullptr, 0, Crypto::MbedTls::CryptoSecurePrng, nullptr);
+#else
+            int rv = mbedtls_pk_parse_key(&mPKey, reinterpret_cast<const unsigned char *>(sSrvKey), sSrvKeyLength,
+                                          nullptr, 0);
+#endif
+            if (rv != 0)
+            {
+                OutputLine("mbedtls_pk_parse_key returned %d", rv);
+            }
+
+            rv = mbedtls_ctr_drbg_seed(&mCtrDrbg, mbedtls_entropy_function, nullptr, nullptr, 0);
+            if (rv != 0)
+            {
+                OutputLine("mbedtls_ctr_drbg_seed returned %d", rv);
+            }
+
+            rv = mbedtls_x509_crt_parse(&mSrvCert, reinterpret_cast<const unsigned char *>(sSrvPem), sSrvPemLength);
+            if (rv != 0)
+            {
+                OutputLine("mbedtls_x509_crt_parse (1) returned %d", rv);
+            }
+            rv = mbedtls_x509_crt_parse(&mSrvCert, reinterpret_cast<const unsigned char *>(sCasPem), sCasPemLength);
+            if (rv != 0)
+            {
+                OutputLine("mbedtls_x509_crt_parse (2) returned %d", rv);
+            }
+            rv = mbedtls_ssl_setup(&mSSLContext, &mSSLConfig);
+            if (rv != 0)
+            {
+                OutputLine("mbedtls_ssl_setup returned %d", rv);
+            }
+        }
+        else if (aArgs[0] == "circular")
```

Comment:
I did it this way initially, but it confuses `clang-format` and causes it to indent the `if` statement by an extra four spaces. That's why I did it as above, with the `#else` clause.

I'm happy to do it the way you suggest if you're OK with the `if` statement being indented by an extra four spaces.

Reply:
I'd tend to the solution with less lines of code, but not strong opinion here.

Reply:
I realized that this can be cleanly resolved by changing the order of the `if`/`else if` clauses so that TLS is not the first. This way, we can avoid using `#else` without confusing `clang-format`. I included this in my latest update.